assembly/src/assembler/mod.rs

```diff
@@ -47,7 +48,7 @@ impl Assembler {
     pub fn new() -> Self {
         Self {
             kernel: Kernel::default(),
-            module_provider: Box::new(()),
+            module_provider: Default::default(),
             proc_cache: Default::default(),
             in_debug_mode: false,
```

Comment:
comment for the future: we can probably get rid of `Assembler::new()` constructor as it is basically the same as the `default()` constructor.

Reply:
Sounds good! Updating on the serialization PR